### PR TITLE
Mongo password reset routine + a bit of leftover changes related to mongo and CI

### PR DESCRIPTION
**Pull request by mo-dkrz**
In this PR we are going to introduce a new routine in the `_setup_mongo` to restart the newly defined password when we are deploying the Freva instance with the new password for the MongoDB

Because CI was red, and as the version of upload and download artifact were deprecated i upgraded to the version 3

### CODE REVIEW

File: assets/share/freva/deployment/scripts/rootless-entrypoint.sh [hunk 1]
```diff
@@ -7,11 +7,21 @@ _setup_mariadb() {
...
     DIR_OWNER=$(stat -c '%U' "/data/configdb")
     if [ "$DIR_OWNER" != "root" ]; then
         chown -R root:0 /data/configdb
     fi
+    echo "starting mongo without authentication"
+    mongod --bind_ip_all --fork
+    echo "creating user ..."
+    mongosh --eval 'use admin; db.createUser({user: "mongo", pwd: "{{root_passwd}}", roles: [{ role: "root", db: "admin" }]});'
+    export MONGO_INITDB_ROOT_USERNAME=mongo
+    export MONGO_INITDB_ROOT_PASSWORD={{root_passwd}}
```
Comment (antarcticrainforest) [inline assets/share/freva/deployment/scripts/rootless-entrypoint.sh:21]: I don't think that the root_passwd variable is replaced. As ansible doens't evaluate this file.

**Review by antarcticrainforest — CHANGES_REQUESTED** [review]
I fear that this approach doesn't work. Because the entrypoint script is only executed if you run the container in rootless mode: See the [playbook](https://github.com/FREVA-CLINT/freva-deployment/blob/1e5903be81b91188a6ddd2c13c4cc6662b78f7b6/assets/share/freva/deployment/playbooks/freva_rest-server-compose-template.yml#L32)

**Comment by mo-dkrz** [comment]
@antarcticrainforest if it's possible could you please review the new approach again. this one is more concise and I've already deployed on a real env and it successfully flush the mongo privileges to the pass. 
Btw since validator vol of the `user-data` was only one single line, I made the change in this branch which in principal is acting as a validator entrypoint against new introduced facets by user. If you disagree i can dismantle it from this branch and submit new PR for this.

**Comment by mo-dkrz** [comment]
ähm, there is an issue in image generator pipeline regarding the newest updates of `debian:bullseye` which it doesn't support the mentioned archs in the [build_job ci](https://github.com/FREVA-CLINT/freva-deployment/blob/main/.github/workflows/build_job.yml#L288-L301).

```bash
❯ docker buildx imagetools inspect debian:bullseye
Name:      docker.io/library/debian:bullseye
MediaType: application/vnd.docker.distribution.manifest.list.v2+json
Digest:    sha256:d0036be35fe0a4d2649bf074ca467a37dab8c5b26bbbdfca0375b4dc682f011d
           
Manifests: 
  Name:      docker.io/library/debian:bullseye@sha256:16f0c16160de30e40a408a6e940083bc1b409fe2a7db93bb0b04262a6ef73419
  MediaType: application/vnd.docker.distribution.manifest.v2+json
  Platform:  linux/amd64
             
  Name:      docker.io/library/debian:bullseye@sha256:6e3a19b773c17e0e5cdb12d96213eb6fd8f21b15211a41d80af571e7d3e4212e
  MediaType: application/vnd.docker.distribution.manifest.v2+json
  Platform:  linux/arm/v7
             
  Name:      docker.io/library/debian:bullseye@sha256:9474a8086a501d13c0ec19aa8059b20c7ed3ed7809955ac46ee865e9e12037dd
  MediaType: application/vnd.docker.distribution.manifest.v2+json
  Platform:  linux/arm64/v8
             
  Name:      docker.io/library/debian:bullseye@sha256:27f356295fe8cb29acbf4c100ba78bcff9e2f6d87ac6b898bc29773602fa2a74
  MediaType: application/vnd.docker.distribution.manifest.v2+json
  Platform:  linux/386
```

But `debian:bookworm` still supports more archs, so switching to bookworm might be the solution

```bash
❯ docker buildx imagetools inspect debian:bookworm
Name:      docker.io/library/debian:bookworm
MediaType: application/vnd.docker.distribution.manifest.list.v2+json
Digest:    sha256:e11072c1614c08bf88b543fcfe09d75a0426d90896408e926454e88078274fcb
           
Manifests: 
  Name:      docker.io/library/debian:bookworm@sha256:bfee693abf500131d9c2aea2e9780a4797dc3641644bac1660b5eb9e1f1e3306
  MediaType: application/vnd.docker.distribution.manifest.v2+json
  Platform:  linux/amd64
             
  Name:      docker.io/library/debian:bookworm@sha256:3aeebec185dad93ead7c3c593e59520219e2b0a4c4f51ba9975a58682a860d5e
  MediaType: application/vnd.docker.distribution.manifest.v2+json
  Platform:  linux/arm/v5
             
  Name:      docker.io/library/debian:bookworm@sha256:7e9ac11ac1b596a03b243104325e1bfcc3feca2f6b3d99f85333e0b6469dae2f
  MediaType: application/vnd.docker.distribution.manifest.v2+json
  Platform:  linux/arm/v7
             
  Name:      docker.io/library/debian:bookworm@sha256:e26c1d5f2e7681a068e30be859417e4cb0250bdc20a53c3b833d809ca0aab06c
  MediaType: application/vnd.docker.distribution.manifest.v2+json
  Platform:  linux/arm64/v8
             
  Name:      docker.io/library/debian:bookworm@sha256:0aac619fdad689ae808651423adea7b7c32ea72a9f8241e9489c3320440471b1
  MediaType: application/vnd.docker.distribution.manifest.v2+json
  Platform:  linux/386
             
  Name:      docker.io/library/debian:bookworm@sha256:8c4202e3022d1e1b250e0ec18f619922efa85a3a345f0989d1973ebbf1bb04fc
  MediaType: application/vnd.docker.distribution.manifest.v2+json
  Platform:  linux/mips64le
             
  Name:      docker.io/library/debian:bookworm@sha256:8d483ba3685b8e201fb2611af4c19601da49630c42b7058aad230104b28975db
  MediaType: application/vnd.docker.distribution.manifest.v2+json
  Platform:  linux/ppc64le
             
  Name:      docker.io/library/debian:bookworm@sha256:3fec0a46444f71429a665428f5d2e535a14e6cabddab21f44277989c114e31b3
  MediaType: application/vnd.docker.distribution.manifest.v2+json
  Platform:  linux/s390x
```

What do you think of switching to bookworm? Or do you have another idea? @antarcticrainforest

**Review by antarcticrainforest — APPROVED** [review]
I haven't looked into it in great detail. But if it works then it works.

LGTM

**Comment by antarcticrainforest** [comment]
> ähm, there is an issue in image generator pipeline regarding the newest updates of `debian:bullseye` which it doesn't support the mentioned archs in the [build_job ci](https://github.com/FREVA-CLINT/freva-deployment/blob/main/.github/workflows/build_job.yml#L288-L301) anymore.
> 
> ```shell
> ❯ docker buildx imagetools inspect debian:bullseye
> Name:      docker.io/library/debian:bullseye
> MediaType: application/vnd.docker.distribution.manifest.list.v2+json
> Digest:    sha256:d0036be35fe0a4d2649bf074ca467a37dab8c5b26bbbdfca0375b4dc682f011d
>            
> Manifests: 
>   Name:      docker.io/library/debian:bullseye@sha256:16f0c16160de30e40a408a6e940083bc1b409fe2a7db93bb0b04262a6ef73419
>   MediaType: application/vnd.docker.distribution.manifest.v2+json
>   Platform:  linux/amd64
>              
>   Name:      docker.io/library/debian:bullseye@sha256:6e3a19b773c17e0e5cdb12d96213eb6fd8f21b15211a41d80af571e7d3e4212e
>   MediaType: application/vnd.docker.distribution.manifest.v2+json
>   Platform:  linux/arm/v7
>              
>   Name:      docker.io/library/debian:bullseye@sha256:9474a8086a501d13c0ec19aa8059b20c7ed3ed7809955ac46ee865e9e12037dd
>   MediaType: application/vnd.docker.distribution.manifest.v2+json
>   Platform:  linux/arm64/v8
>              
>   Name:      docker.io/library/debian:bullseye@sha256:27f356295fe8cb29acbf4c100ba78bcff9e2f6d87ac6b898bc29773602fa2a74
>   MediaType: application/vnd.docker.distribution.manifest.v2+json
>   Platform:  linux/386
> ```
> 
> But `debian:bookworm` still supports more archs, so switching to bookworm might be the solution
> 
> ```shell
> ❯ docker buildx imagetools inspect debian:bookworm
> Name:      docker.io/library/debian:bookworm
> MediaType: application/vnd.docker.distribution.manifest.list.v2+json
> Digest:    sha256:e11072c1614c08bf88b543fcfe09d75a0426d90896408e926454e88078274fcb
>            
> Manifests: 
>   Name:      docker.io/library/debian:bookworm@sha256:bfee693abf500131d9c2aea2e9780a4797dc3641644bac1660b5eb9e1f1e3306
>   MediaType: application/vnd.docker.distribution.manifest.v2+json
>   Platform:  linux/amd64
>              
>   Name:      docker.io/library/debian:bookworm@sha256:3aeebec185dad93ead7c3c593e59520219e2b0a4c4f51ba9975a58682a860d5e
>   MediaType: application/vnd.docker.distribution.manifest.v2+json
>   Platform:  linux/arm/v5
>              
>   Name:      docker.io/library/debian:bookworm@sha256:7e9ac11ac1b596a03b243104325e1bfcc3feca2f6b3d99f85333e0b6469dae2f
>   MediaType: application/vnd.docker.distribution.manifest.v2+json
>   Platform:  linux/arm/v7
>              
>   Name:      docker.io/library/debian:bookworm@sha256:e26c1d5f2e7681a068e30be859417e4cb0250bdc20a53c3b833d809ca0aab06c
>   MediaType: application/vnd.docker.distribution.manifest.v2+json
>   Platform:  linux/arm64/v8
>              
>   Name:      docker.io/library/debian:bookworm@sha256:0aac619fdad689ae808651423adea7b7c32ea72a9f8241e9489c3320440471b1
>   MediaType: application/vnd.docker.distribution.manifest.v2+json
>   Platform:  linux/386
>              
>   Name:      docker.io/library/debian:bookworm@sha256:8c4202e3022d1e1b250e0ec18f619922efa85a3a345f0989d1973ebbf1bb04fc
>   MediaType: application/vnd.docker.distribution.manifest.v2+json
>   Platform:  linux/mips64le
>              
>   Name:      docker.io/library/debian:bookworm@sha256:8d483ba3685b8e201fb2611af4c19601da49630c42b7058aad230104b28975db
>   MediaType: application/vnd.docker.distribution.manifest.v2+json
>   Platform:  linux/ppc64le
>              
>   Name:      docker.io/library/debian:bookworm@sha256:3fec0a46444f71429a665428f5d2e535a14e6cabddab21f44277989c114e31b3
>   MediaType: application/vnd.docker.distribution.manifest.v2+json
>   Platform:  linux/s390x
> ```
> 
> What do you think of switching to bookworm? Or do you have another idea? @antarcticrainforest

I think that is an option. I am not sure about the python version though. But you can give it a shot.

**Comment by antarcticrainforest** [comment]
The [build pipeline has failed](https://github.com/FREVA-CLINT/freva-deployment/actions/runs/11506406208/job/32030272070). I've just checked and I think you want to used this tag: [bullseye-20240904](https://hub.docker.com/layers/library/debian/bullseye-20240904/images/sha256-2a1c9a0e7aa4fcad4d6319547bfcc0d8a7a290fada438d6fabc416f137cf12b4?context=explore)

**Pull request merged**